packages/aries-hooks/src/index.tsx

```diff
@@ -0,0 +1,295 @@
+import React, { createContext, useState, useEffect, useContext } from 'react'
+
+import { downloadGenesis, storeGenesis } from '../genesis-utils'
+
+import {
+  Agent,
+  InitConfig,
+  HttpOutboundTransport,
+  ConnectionState,
+  CredentialState,
+  ProofState,
+  ConnectionEventTypes,
+  CredentialEventTypes,
+  ProofEventTypes,
+  ConnectionStateChangedEvent,
+  CredentialStateChangedEvent,
+  ProofStateChangedEvent,
+  ConnectionRecord,
+  ProofRecord,
+  CredentialRecord,
+  WsOutboundTransport,
+  BasicMessageRecord,
+  BasicMessageReceivedEvent,
+  BasicMessageEventTypes,
+} from '@aries-framework/core'
+
+import { agentDependencies } from '@aries-framework/react-native'
+
+const TAG = 'Aries-Hooks:'
+
+const AgentContext = createContext<any>({})
+const ConnectionContext = createContext<any>({})
+const CredentialContext = createContext<any>({})
+const ProofContext = createContext<any>({})
+const BasicMessageContext = createContext<any>({})
+
+// Agent
+export const useAgent = (): { agent: Agent; loading: boolean } => {
+  return useContext(AgentContext)
+}
+
+// Connection
+export const useConnections = (): { connections: ConnectionRecord[]; loading: boolean } => {
+  return useContext(ConnectionContext)
+}
+
+export const useConnectionById = (id: string): ConnectionRecord | undefined => {
+  const { connections } = useContext(ConnectionContext)
+  const connection = connections.find((c: ConnectionRecord) => c.id === id)
+  if (!connection) {
+    console.warn(TAG, `Could not find Connection By ID ${id}`)
```

Comment:
No `console.xxx` please. Maybe we could use the agent logger or just return a null.